HIP/hip-857.md

```diff
@@ -0,0 +1,236 @@
+- As a mirror node operator, I want to provide my API clients with the ability to list NFT allowances for a spender who has received an allowance for a single NFT serial number, as well as via `approveForAll`.
+
+## Specification
+
+This is a HIP for adding the ability to query the allowances on NFTs from the mirror node.
+
+### Constraints
+
+When designing the NFT allowances APIs, it is important to consider certain constraints on the number of allowances and tokens. These constraints may limit the ability to query unbounded information.
+
+- An approval transaction can include a maximum of 20 allowance change operations.
+- An account can have a maximum of 100 hbar, token, and approved for all NFT allowances.
+- An account can have an unbounded number of individual NFT allowances.
+- A non-fungible token can have an unbounded number of approved for all NFT allowances.
+- A specific NFT can only have one explicit approval. The last one to be approved will take precedence.
+- When an NFT is transferred by either the owner or a spender, it results in the individual spender's allowance being cleared.
+- A specific NFT can have an unbounded number of implicit approved for all NFT allowances.
+
+### Approved for all NFT Allowances API
+
+This API accepts a path parameter that represents either the owner or spender, depending on a boolean flag provided as a query parameter called `owner`. When the `owner` value is true, the `accountId` path parameter should specify the ID of the owner, and the API will retrieve the allowances that the owner has granted to different spenders. Conversely, when the `owner` value is false, the `accountId` path parameter should indicate the ID of the spender who has an allowance, and the API will instead provide the allowances granted to the spender by different owners of those tokens.
+
+Following are example responses of the NFT allowance API:
+
+GET `/api/v1/accounts/1000/allowances/nfts?owner=true`
+
+```json
+{
+  "allowances": [
+    {
+      "approved_for_all": true,
+      "owner": "0.0.1000",
+      "spender": "0.0.8488",
+      "token_id": "0.0.1033",
+      "timestamp": {
+        "from": "1633466229.96874612",
+        "to": null
+      }
+    },
+    {
+      "approved_for_all": false,
+      "owner": "0.0.1000",
+      "spender": "0.0.8489",
+      "token_id": "0.0.1034",
+      "timestamp": {
+        "from": "1633466229.96874618",
+        "to": null
+      }
+    },
+    {
+      "approved_for_all": true,
+      "owner": "0.0.1000",
+      "spender": "0.0.9857",
+      "token_id": "0.0.1032",
+      "timestamp": {
+        "from": "1633466229.96884615",
+        "to": null
+      }
+    }
+  ],
+  "links": {
+		"next": "/api/v1/accounts/0.0.1000/allowances/nfts?limit=3&order=asc&account.id=gte:9857&token.id=gt:0.0.1034"
```

Comment:
limit=3 would have to be in the initial URL for this link to be accurate. Same for other example.